drivers/power_domain/power_domain_intel_adsp.c

```diff
@@ -24,6 +24,9 @@ static int pd_intel_adsp_set_power_enable(struct pg_bits *bits, bool power_enabl
 	if (power_enable) {
 		sys_write16(sys_read16((mem_addr_t)&ACE_DfPMCCU.dfpwrctl) | SPA_bit_mask,
 			    (mem_addr_t)&ACE_DfPMCCU.dfpwrctl);
+		while ((sys_read16((mem_addr_t)&ACE_DfPMCCU.dfpwrsts) & BIT(bits->CPA_bit)) == 0) {
```

Comment:
WAIT_FOR

Reply:
I'm not sure if that's a good solution. There is no scenario where we are waking-up a power domain and it stays off. This would mean a hardware defect, how should we handle it?

Reply:
This api support errors, device will be not available and application will send error back to host when trying to start gpdma, thats fine for me. 
Anyway not a big deal because of super rare case